### PR TITLE
Home task 2

### PR DESCRIPTION
**Pull request by antonfilichkin**
Home task 2

### CODE REVIEW

**Review by alpakhomov — APPROVED** [review]
Approved!

**Review by MChislov — APPROVED** [review]
I'd prefer to see your own logic in the first task. But proposed solution does work.

**Comment by antonfilichkin** [comment]
> I'd prefer to see your own logic in the first task. But proposed solution does work.

@MChislov Updated the first one with my own logic

**Comment by MChislov** [comment]
@antonfilichkin thank you.
I'm afraid your solution does not work.
For this array [3, 3, 3, 3, 3, 3, 1, 8, 9, 1, 9] a new function result seems to be (3, 8) expected. Your solution gives (9, 1).
Please correct me if I'm wrong.

**Comment by antonfilichkin** [comment]
> @antonfilichkin thank you. I'm afraid your solution does not work. For this array [3, 3, 3, 3, 3, 3, 1, 8, 9, 1, 9] a new function result seems to be (3, 8) expected. Your solution gives (9, 1). Please correct me if I'm wrong.

@MChislov Thanks! Yep. My bad :( Fixed

File: ht2/hw1_1.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+def major_and_minor_elem(inp: List) -> Tuple[int, int]:
+    counter = Counter(inp).most_common()
+    return counter[0][0], counter[-1][0]
+
+
+def major_and_minor_elem_2(inp: List) -> Tuple[int, int]:
+    elements_count = {}
+    for element in inp:
+        elements_count[element] = elements_count.get(element, 0) + 1
```
Comment (MChislov) [inline ht2/hw1_1.py:32]:
```suggestion
        elements_count[element]+=1
```

Comment (MChislov) [reply ht2/hw1_1.py:32]: Solution looks fine but I'd suggest this change

**Review by MChislov — APPROVED** [review]

File: ht2/hw1_1.py (hunk 1 above)
Comment (antonfilichkin) [reply ht2/hw1_1.py:32]: But then we need some logic to set 0 for the dict value in case the key is not present?

Comment (antonfilichkin) [inline ht2/hw1_1.py:32]: Something like this then?
```suggestion
        elements_count.setdefault(element, 0)
        elements_count[element] += 1
```

Comment (MChislov) [reply ht2/hw1_1.py:32]: absolutely. Thanks.

Comment (MChislov) [reply ht2/hw1_1.py:32]: You are right. setdefault makes the same as your logic. Thank you

**Pull request merged**